Ermish/vue3refresh

ermish opened the pull request:
# Site is alive with vue3/nuxt3!
Resolves #30, #23, #24, #25
- Upgraded to Vue3
- Upgraded to Nuxt3
- Upgraded to Yarn 3.2.2 (berry)
- Remove imcompatible google analytics nuxt plugin (for now)
- Created Home/Search page, about, news, login, admin dashboard empty templates.
- migrated components to vue3 
- added pinia for state management
- Adding in eslint, stylelint, editorconfig
- updated tsconfig to support vue3
- Updated yarn build commands 
- Added min node version dependency for builds

theyokohamalife commented:
@ermish I tried to run this using `yarn dev` but I'm getting the following error. It looks like it's being caused by Nitro?
![Screen Shot 2022-07-21 at 11 59 39 PM](https://user-images.githubusercontent.com/31802656/180246668-aaf3b7ed-56ad-4367-bf7e-38a601bc4d73.png)

theyokohamalife commented:
It seems that the `Vue.extend()` you're using in the LoginPage and AdminDashboardPage are causing the issue. I removed those and was finally able to get the server to run. Is this block of code needed?

```
<script lang="ts">
import Vue from 'vue'

export default Vue.extend({
    name: 'IndexPage'
})
</script>
```

theyokohamalife commented:
@ermish I'd like to see a preview of this on Netlify before it gets merged to main. Can you re-enable previews for PRs?

ermish commented:
oh wierd! I updated that 🤔  perhaps I have a local commit I didn't check in


> It seems that the `Vue.extend()` you're using in the LoginPage and AdminDashboardPage are causing the issue. I removed those and was finally able to get the server to run. Is this block of code needed?
> 
> ```
> <script lang="ts">
> import Vue from 'vue'
> 
> export default Vue.extend({
>     name: 'IndexPage'
> })
> </script>
> ```

ermish commented:
> @ermish I'd like to see a preview of this on Netlify before it gets merged to main. Can you re-enable previews for PRs?

finally got them working! Had to upgrade the netlify build image and node version dependency. 
Here's the new preview
https://deploy-preview-29--findadoc.netlify.app/

theyokohamalife commented:
@ermish Had a look at the preview and noticed that there are links to the empty pages in the top app bar. Can those be removed until they're ready and we've decided on a navigation system? Also, I think we can remove the News component altogether, since it won't be included in the MVP.

theyokohamalife commented:
@ermish I'm still unable to run the project. I'm now getting the following message after `yarn install` and `yarn dev`: 
<img width="1096" alt="Screen Shot 2022-07-24 at 3 49 24 PM" src="https://user-images.githubusercontent.com/31802656/180635949-358faaf3-b319-4701-a87d-5652d39de858.png">

ermish commented:
> @ermish I'm still unable to run the project. I'm now getting the following message after `yarn install` and `yarn dev`: 
> 
> <img width="1096" alt="Screen Shot 2022-07-24 at 3 49 24 PM" src="https://user-images.githubusercontent.com/31802656/180635949-358faaf3-b319-4701-a87d-5652d39de858.png">
> 
> 

Hmmm not sure why. That seems to be a package issue 🤔 we might need to walk through this one

ermish commented:
> @ermish Had a look at the preview and noticed that there are links to the empty pages in the top app bar. Can those be removed until they're ready and we've decided on a navigation system? 
> 
> 
> 
> <img width="1439" alt="Screen Shot 2022-07-24 at 3 38 15 PM" src="https://user-images.githubusercontent.com/31802656/180635838-b05b727d-ea67-45fc-b55f-65c70de251dc.png">
> 
> 

For now, I just want to get this in here so I can start developing again. Since these aren't going out to production I didn't want to add temporary logic for hiding the navbar. There will be a whole separate ticket for working on the navbar section and the welcome banner. I just wanted to verify the vue3/nuxt3 routing is working and have an example

ermish commented:
@theyokohamalife can we get this merged in and I'll investigate the build in the next PR?

theyokohamalife commented:
> @theyokohamalife can we get this merged in and I'll investigate the build in the next PR?

I understand that you want to get this merged quickly, but also please understand that I can't approve it if I can't get it working.

It might help if you break this PR down into smaller ones, because it's trying to do a lot of things and it's hard to pinpoint where things are going wrong.

ann-kilzer commented on .eslintrc.js:
```diff
@@ -0,0 +1,111 @@
+    // 'off' or 0 - turn the rule off
+    // 'warn' or 1 - turn the rule on as a warning (doesn’t affect exit code)
+    // 'error' or 2 - turn the rule on as an error (exit code will be 1)
+    rules: {
```
Might it be better for us to adopt a set of rules (like airbnb) that we can use across our different repos, rather than hand-configuring different rules?

ermish replied:
For now, I just used my custom set of rules to get started. IMO there are some gaps in most of the default packages such as Airbnb.
We should walk through linting rules in our next dev meeting 👍

ermish replied:
After we discuss, I can update them in a future commit

ann-kilzer commented on pages/newsPage.vue:
```diff
@@ -0,0 +1,14 @@
+<template>
```
I think we should skip this one for MVP

ermish replied:
I just added it as an empty page to test out the navigation. 
I'll remove it

ann-kilzer commented on components/Welcome.vue:
```diff
@@ -1,52 +1,20 @@
+        <a href="https://info.findadoc.jp">
+            <button
+                type="button"
+                class="my-8 inline-flex items-center px-6 py-3 border border-transparent text-base font-medium rounded-md shadow-sm text-white bg-indigo-600 hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
```
Can we set the linter to check line width? This line is too long

ermish replied:
Yeah I haven't run the linter across the project or adding linting automation yet 👍 will fix in future linter setup commit #42

ermish commented on pages/admindashboardPage.vue:
```diff
@@ -0,0 +1,14 @@
+<template>
```
Good find, updating.

ermish commented:
@ann-kilzer updated! please check  :)

ermish commented:
@theyokohamalife I just upgraded the yarn version in the project to the latest 2+. See if it helps the project build for you now. 

@theyokohamalife @ann-kilzer 
To check your yarn version, use `yarn -v` and you can upgrade to the latest using `yarn set version stable`

theyokohamalife commented:
> @theyokohamalife I just upgraded the yarn version in the project to the latest 2+. See if it helps the project build for you now.
> 
> @theyokohamalife @ann-kilzer To check your yarn version, use `yarn -v` and you can upgrade to the latest using `yarn set version stable`

@ermish I'm using yarn `3.2.2`

ermish commented:
> > @theyokohamalife I just upgraded the yarn version in the project to the latest 2+. See if it helps the project build for you now.
> > @theyokohamalife @ann-kilzer To check your yarn version, use `yarn -v` and you can upgrade to the latest using `yarn set version stable`
> 
> @ermish I'm using yarn `3.2.2`

@theyokohamalife ok we should be good to merge! Please check if it works for you now

theyokohamalife commented on .editorconfig:
```diff
@@ -0,0 +1,9 @@
+[*]
+insert_final_newline = true
+indent_style = space
+indent_size = 4
```
I'd prefer to remove the indentation changes now instead of merging them into main and having to undo them.

ermish replied:
at this point it will be unnecessary work. Most of these files are new and created in 4, so there's no undo-ing.
 I'm hoping to make the case for 4 instead of 2. :)

theyokohamalife commented on .eslintrc.js:
```diff
@@ -0,0 +1,111 @@
+        'function-paren-newline': ['error', 'consistent'],
+        indent: [
+            2,
+            4,
```
Same here and other places.

theyokohamalife commented on .stylelintrc.json:
```diff
@@ -0,0 +1,41 @@
+        "color-hex-length": "short",
+        "color-named": "never",
+        "comment-empty-line-before": null,
+        "indentation": 4,
```
Here as well. I'm probably missing some lines.

theyokohamalife commented on components/TopNav.vue:
```diff
@@ -1,16 +1,32 @@
-  </div>
 </template>
 
+<script lang="ts">
+import { defineComponent } from 'vue'
+
+export default defineComponent({
+    props: {
+        msg: {
+            type: String,
+            required: false
+        }
+    },
+    data() {
+        return {
+            name: '',
+        }
+    }
+})
+</script>
```
I don't want to check these `<script>` sections in for a few reasons.

1. They aren't currently being used and are just adding unnecessary lines to the code.
2. They might encourage people to start importing and exporting components the Vue way instead of the Nuxt way, which defeats one of the biggest reasons for using Nuxt.

For now, please remove these. 🙏🏾

ermish replied:
these files are literally for my personal use at the moment and I'm going to be filling out a lot of logic in these script sections. For now these are placeholders that I'm actively filling out locally until this can get merged in. 

This PR is just templating, not final files.

theyokohamalife commented on pages/aboutPage.vue:
```diff
@@ -0,0 +1,19 @@
+    </div>
+</template>
+
+<script lang="ts">
+import { defineComponent } from 'vue'
+export default defineComponent({
+})
+</script>
```
```suggestion

```

ermish replied:
Is there a difference in the suggestion? 😅  I can't tell

ermish replied:
ohhhh...it's to remove it....I'm literally re-adding them in in future commits. For now, we should leave them. Especially considering I'm the sole developer on these for now. Before we have other contributors work on them, the files will be completely different

theyokohamalife commented on pages/adminDashboardPage.vue:
```diff
@@ -0,0 +1,15 @@
+    </div>
+</template>
+
+<script lang="ts">
+import { defineComponent } from 'vue'
+export default defineComponent({
+    
+})
+</script>
```
```suggestion

```

theyokohamalife commented on pages/index.vue:
```diff
@@ -1,20 +1,29 @@
-
-export default Vue.extend({
-  name: "IndexPage",
-});
+import { defineComponent } from 'vue'
+export default defineComponent({
+    props: {
+        name: String,
+        msg: { type: String, required: true }
+    },
+    data() {
+        return {
+            count: 1
+        }
+    },
+    mounted() {
+    }
+})
 </script>
```
Please remove 🙏🏾

theyokohamalife commented on pages/loginPage.vue:
```diff
@@ -0,0 +1,14 @@
+    </div>
+</template>
+
+<script lang="ts">
+import { defineComponent } from 'vue'
+export default defineComponent({
+})
+</script>
```
```suggestion

```

theyokohamalife commented on components/TopNav.vue:
```diff
@@ -1,16 +1,32 @@
+        <div>
+            <NuxtLink to="/">Home</NuxtLink>
+            <NuxtLink to="/aboutPage">About</NuxtLink>
+            <NuxtLink to="/newsPage">News</NuxtLink>
```
```suggestion
           
```

theyokohamalife replied:
This link should be remove since the `newsPage` has been deleted.

ermish replied:
good find, updated
